shell/server/core.js

```diff
@@ -116,8 +116,16 @@ NotificationHandle.prototype.close = function () {
   });
 };
 
-saveFrontendRef = function (frontendRef, owner, userId) {
+saveFrontendRef = function (frontendRef, owner, creatorUserId, parentToken, grainId) {
   return inMeteor(function () {
+    if (parentToken) {
+      var ret = makeChildTokenInternal(parentToken, owner, [], grainId);
+      return {sturdyRef: ret.token};
+    }
+    var requirement = {
+      userIsAdmin: creatorUserId
```

Comment:
This seems pretty confusing: `saveFrontendRef()` takes a parameter called `creatorUserId`, but we interpret it as "this user is required to be admin". Do we even want such a requirement in all cases? It seems like the parameter should be a requirement list, and the caller should decide what requirements to pass.